q2_demux/plugin_setup.py

```diff
@@ -228,4 +230,39 @@
                  'sequences after subsampling.')
 )
 
+T = TypeMatch([SequencesWithQuality, PairedEndSequencesWithQuality,
+               JoinedSequencesWithQuality])
+plugin.methods.register_function(
+    function=q2_demux.filter_samples,
+    inputs={'samples': SampleData[T]},
```

Comment:
Thinking about the API here, `demux` is a pretty common shorthand for this data, so maybe we should name it `demux` instead of `samples` (which would mean updating this name elsewhere).